src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,186 @@
+import com.internet.shop.model.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT id, name, login, password FROM users WHERE deleted = FALSE "
+                + "AND login = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            User user = new User();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+                user.setRoles(getUsersRoles(user));
+            }
+            return Optional.of(user);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find user with login " + login, e);
+        }
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users(name, login, password) VALUES (?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create user " + user.toString(), e);
+        }
+        createUsersRole(user);
+        return user;
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        String query = "SELECT id, name, login, password FROM users WHERE deleted = FALSE "
+                + "AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            User user = new User();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+                user.setRoles(getUsersRoles(user));
+            }
+            return Optional.of(user);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find user with id " + id, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't get user with id " + id, e);
```
maybe there is user with such id, but something go wrong